### PR TITLE
Remove DELETE from the LDPCv OPTIONS header

### PR DESCRIPTION
**Pull request by robyj**
- remove the DELETE option from the OPTIONS returned to a user

Resolves: https://jira.duraspace.org/browse/FCREPO-2920

**The title of this pull-request should be a brief description of what the pull-request fixes/improves/changes. Ideally 50 characters or less.**
* * *

**JIRA Ticket**: (link)

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
A brief description of what the intended result of the PR will be and/or what problem it solves.

# What's new?
A in-depth description of the changes made by this PR. Technical details and possible side effects.

Example:
* Changes x feature to such that y
* Added x
* Removed y

# How should this be tested?

A description of what steps someone could take to:
* Reproduce the problem you are fixing (if applicable)
* Test that the Pull Request does what is intended.
* Please be as detailed as possible.
* Good testing instructions help get your PR completed faster.

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

Example:
* Does this change require documentation to be updated? 
* Does this change add any new dependencies? 
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? 
* Could this change impact execution of existing code?

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo4/committers

### CODE REVIEW

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java [hunk 1]
```diff
@@ -578,7 +578,7 @@ protected void addLinkAndOptionsHttpHeaders(final FedoraResource resource) {
         if (resource.isMemento()) {
             options = "GET,HEAD,OPTIONS,DELETE";
         } else if (resource instanceof FedoraTimeMap) {
-            options = "POST,HEAD,GET,OPTIONS,DELETE";
+            options = "POST,HEAD,GET,OPTIONS";
```
Comment (awoods) [inline fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java]: This looks good.
Could you additionally improve the "[FedoraVersioningIT.testOptionsTimeMap](https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java#L1471-L1479)" test to check for a specific list of OPTIONS values?

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java [hunk 2]
```diff
@@ -850,6 +850,8 @@ private static void verifyTimeMapHeaders(final CloseableHttpResponse response, f
         checkForLinkHeader(response, uri + "/" + FCR_VERSIONS, "timemap");
         checkForLinkHeader(response, VERSIONING_TIMEMAP_TYPE, "type");
         checkForLinkHeader(response, ldpcvUri + "/" + FCR_ACL, "acl");
+        assertFalse(response.getFirstHeader("Allow").getValue().contains("DELETE"));
+        assertTrue(response.getFirstHeader("Allow").getValue().contains("POST,HEAD,GET"));
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java]: Would you mind checking these individually... it seems a little fragile to expect the exact string of `POST,HEAD,GET`.
Thanks!

Comment (robyj) [reply fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java]: I can, I thought it'd be OK as the options are held as a fixed string, so the string wouldn't change

Comment (awoods) [reply fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java]: If you could make the change, it would be helpful.

**Pull request merged**